samples/corda/corda-simple-application/clients/src/main/kotlin/com/cordaSimpleApplication/client/AssetStateManager.kt

```diff
@@ -0,0 +1,284 @@
+    val config by requireObject<Map<String, String>>()
+    override fun run() {
+        issueAssetStateHelper(quantity.toInt(), tokenType, config)
+    }
+}
+
+/**
+ * Helper function used by IssueAssetStateCommand
+ */
+fun issueAssetStateHelper(numberOfTokens: Int, tokenType: String, config: Map<String, String>) {
+    val rpc = NodeRPCConnection(
+            host = config["CORDA_HOST"]!!,
+            username = "clientUser1",
+            password = "test",
+            rpcPort = config["CORDA_PORT"]!!.toInt())
+    try {
+        println("IssueAsset flow with arguments $numberOfTokens, $tokenType")
+        val proxy = rpc.proxy
+        val createdState = proxy.startFlow(::IssueAssetState, numberOfTokens, tokenType)
+                .returnValue.get().tx.outputStates.first() as AssetState
+        println(createdState)
+    } catch (e: Exception) {
+        println(e.toString())
+    } finally {
+        rpc.close()
+    }
+}
+
+class IssueAssetStateFromStateRefCommand : CliktCommand(help = "Invokes the IssueAssetStateFromStateRef flow. Requires a linearId") {
+    private val linearId: String by argument()
+    val config by requireObject<Map<String, String>>()
+    override fun run() {
+        val rpc = NodeRPCConnection(
+            host = config["CORDA_HOST"]!!,
+            username = "clientUser1",
+            password = "test",
+            rpcPort = config["CORDA_PORT"]!!.toInt())
+        try {
+            println("IssueAsset flow with arguments $linearId")
+            val proxy = rpc.proxy
+            val createdState = proxy.startFlow(::IssueAssetStateFromStateRef, linearId)
+                .returnValue.get().tx.outputStates.first() as AssetState
+            println(createdState)
+        } catch (e: Exception) {
+            println(e.toString())
+        } finally {
+            rpc.close()
+        }
+    }
+}
+
+/**
+ * The CLI command used to trigger a GetStateByTokenType flow.
+ *
+ * @property tokenType The filter criteria for the [AssetState]s to be retrieved.
+ */
+class GetAssetStatesByTypeCommand : CliktCommand(help = "Get asset states by token type. Requires a token type") {
+    val tokenType: String by argument()
+    val config by requireObject<Map<String, String>>()
+    override fun run() {
+        println("Get states with type $tokenType")
+        val rpc = NodeRPCConnection(
+            host = config["CORDA_HOST"]!!,
+            username = "clientUser1",
+            password = "test",
+            rpcPort = config["CORDA_PORT"]!!.toInt())
+        try {
+            val proxy = rpc.proxy
+            val states = proxy.startFlow(::GetStatesByTokenType, tokenType)
+                .returnValue.get()
+            println(states.toString(Charsets.UTF_8))
+        } catch (e: Exception) {
+            println(e.toString())
+        } finally {
+            rpc.close()
+        }
+    }
+}
+
+/**
+ * The CLI command used to trigger a GetStateByLinearId flow.
```

Comment:
Same comment as above.

Reply:
Name mentioned here is the name of the flow called by this function.